packages/application-shell/src/components/application-shell/application-shell.tsx

```diff
@@ -149,6 +155,17 @@ export const RestrictedApplication = <
   // However, every route change will trigger a re-render. This is probably
   // ok-ish but we might want to look into a more performant solution.
   const location = useLocation();
+
+  const notificationsGlobalRef = useRef<HTMLDivElement>(null);
+  const notificationsPageRef = useRef<HTMLDivElement>(null);
+  const layoutRefs = useRef<{
+    notificationsGlobalRef: RefObject<HTMLDivElement>;
+    notificationsPageRef: RefObject<HTMLDivElement>;
+  }>({
+    notificationsGlobalRef,
+    notificationsPageRef,
+  });
```

Comment:
I tried different things but eventually ended up with this.

We're "gathering" different refs into one and pass it to the `PortalsContainer`.